test/integration/spec/logger.js

```diff
@@ -0,0 +1,101 @@
+/* eslint-disable no-console, no-undefined */
+'use strict';
+
+const assert = require('assert');
+const sinon = require('sinon');
+
+const { Logger } = require('../../../lib');
+const defaults = require('../../lib/defaults');
+const connect = require('../../../lib/connect');
+const getToken = require('../../lib/token');
+const { createRoom } = require('../../lib/rest');
+const { randomName } = require('../../lib/util');
+
+describe('logger', function() {
+  // eslint-disable-next-line no-invalid-this
+  this.timeout(60000);
+  const DEFAULT_LOGGER_NAME = 'twilio-video';
+
+  let logger;
+  let loggerCb;
+  let sid;
+  let token;
+
+  const loadPlugin = (loggerName, showLogs) => {
+    loggerCb = sinon.spy();
+    loggerName = loggerName || DEFAULT_LOGGER_NAME;
+
+    // Reset logger module cache
+    const loggers = Logger.getLoggers();
+    Object.keys(loggers).forEach(name => delete loggers[name]);
+
+    logger = Logger.getLogger(loggerName);
+    const originalFactory = logger.methodFactory;
+    logger.methodFactory = function(methodName, level, loggerName) {
+      const method = originalFactory(methodName, level, loggerName);
+      return function(datetime, logLevel, component, message, data) {
+        loggerCb({ datetime, logLevel, component, message, data });
+        if (showLogs) {
+          method(...arguments);
+        }
+      };
+    };
+    logger.setLevel('info');
+  };
+
+  beforeEach(async () => {
+    const identity = randomName();
+    token = getToken(identity);
+    sid = await createRoom(randomName(), defaults.topology);
+    loadPlugin();
+  });
+
+  it('should initialize logger using default logger name', async () => {
+    await connect(token, Object.assign({ name: sid }, defaults));
+    sinon.assert.called(loggerCb);
+  });
+
+  it('should initialize logger using custom logger name', async () => {
+    loadPlugin('my-logger');
+    await connect(token, Object.assign({ name: sid, loggerName: 'my-logger' }, defaults));
+    sinon.assert.called(loggerCb);
+  });
+
+  it('should initialize loggers for each individual components', async () => {
+    await connect(token, Object.assign({ name: sid, logLevel: {
+      default: 'error',
+      media: 'warn',
+    } }, defaults));
+
+    assert(!!Logger.getLogger(DEFAULT_LOGGER_NAME + '-default'));
+    assert(!!Logger.getLogger(DEFAULT_LOGGER_NAME + '-media'));
+  });
+
+  it('should provide required arguments via the plugin', async () => {
+    await connect(token, Object.assign({ name: sid }, defaults));
+    const callSpies = loggerCb.getCalls();
+    assert(!!callSpies.length);
+    callSpies.forEach(callSpy => {
+      ['datetime', 'logLevel', 'component', 'message'].forEach(argName => {
+        assert(!!callSpy.args[0]);
+        assert(!!callSpy.args[0][argName]);
+      });
+    });
+  });
+
+  it('should provide optional data via the plugin', async () => {
```

Comment:
I think we need couple of tests to verify that 
the specified loglevel is respected. 
a) That is when logLevel = 'warn' we should not see logs debug logs.  
b) when loglevel is 'debug' we should see 'debug' logs.
I tried b) and did not see debug logs.

Reply:
I think the reason why b) is not working for you is because you previous called `setLevel('warn')` which would save the state to localStorage. Just a reminder, `logLevel` parameter in connect options will be used as the default loglevel. See [here](https://github.com/twilio/twilio-video.js/blob/JSDK-2373/lib/util/log.js#L82). Meaning, if there's no previously saved state, the logger will use the default one. But if there's a previously saved loglevel, logger will use that instead. I added this in the `logLevel` parameter documentation.

Reply:
ah got it! thanks for the explaination - I guess it sounds fine then.